src/component/modal.js

```diff
@@ -34,15 +34,39 @@ const styles = StyleSheet.create({
   },
 });
 
-const Modal = ({ title = '', onClose, children, style }) => (
-  <View style={[styles.modal, style]}>
-    <H4Text style={styles.title}>{title.toUpperCase()}</H4Text>
-    <Button style={styles.cancelBtn} onPress={onClose}>
-      <Icon image="cancel-grey" style={styles.cancelIcon} />
-    </Button>
-    {children}
-  </View>
-);
+class Modal extends Component {
+  constructor(props) {
+    props.title = props.title || '';
+    super(props);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
```

Comment:
Not a fan of the bind style. We should use arrow functions in the event handlers themselves to stay consistent with our current coding style.